src/webpimage.cpp

```diff
@@ -517,6 +517,8 @@ namespace Exiv2 {
             DataBuf payload(size);
 
             if (equalsWebPTag(chunkId, WEBP_CHUNK_HEADER_VP8X) && !has_canvas_data) {
+                enforce(size >= 10, Exiv2::kerCorruptedMetadata);
```

Comment:
I took a look at the following code and checked how large `size` must be at lest so that the `memcpy` calls don't read out of bounds. Unfortunately the values are simply inserted there as numbers. I can take a look whether we can give them meaningful names.